core/app/services/spree/cart/destroy.rb

```diff
@@ -0,0 +1,26 @@
+module Spree
+  module Cart
+    class Destroy
+      prepend Spree::ServiceModule::Base
+
+      def call(order:)
+        run :check_if_can_be_destroyed
```

Comment:
@KacperMekarski we also need to add a step to void all payments :)

Reply:
@damianlegawiec done!